{"error": null, "issues": [{"address": 250, "code": "adr,bytes data) payable public {\n        require(msg.sender == Owner);\n         // <yes> <report> UNCHECKED_LL_CALLS\n        adr.call.value(msg.value)(data);\n    }\n}", "debug": "calldata_HomeyJar_32 + 4: 0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffc\ncalldata_HomeyJar_0: 0xba21d62a00000000000000000000000000000000000000000000000000000000\ncalldatasize_HomeyJar: 0x4\n", "description": "A possible integer overflow exists in the function `Command(address,bytes)`.\nThe addition or multiplication may result in a value higher than the maximum representable integer.", "filename": "/data/demo_analysis/source_code/0x70f9eddb3931491aab1aeafbc1e7f1ca2a012db4.sol", "function": "Command(address,bytes)", "lineno": 26, "title": "Integer Overflow ", "type": "Warning"}, {"address": 683, "code": "lance);\n    }\n    \n    funct", "debug": "", "description": "A possible transaction order independence vulnerability exists in function withdraw(). The value or direction of the call statement is determined from a tainted storage location", "filename": "/data/demo_analysis/source_code/0x70f9eddb3931491aab1aeafbc1e7f1ca2a012db4.sol", "function": "withdraw()", "lineno": 23, "title": "Transaction order dependence", "type": "Warning"}, {"address": 968, "code": "ta);\n    }\n}", "debug": "", "description": "This contract executes a message call to an address provided as a function argument. Generally, it is not recommended to call user-supplied addresses using Solidity's call() construct. Note that attackers might leverage reentrancy attacks to exploit race conditions or manipulate this contract's state.", "filename": "/data/demo_analysis/source_code/0x70f9eddb3931491aab1aeafbc1e7f1ca2a012db4.sol", "function": "Command(address,bytes)", "lineno": 29, "title": "Message call to external contract", "type": "Warning"}, {"address": 968, "code": "ta);\n    }\n}", "debug": "", "description": "The return value of an external call is not checked. Note that execution continue even if the called contract throws.", "filename": "/data/demo_analysis/source_code/0x70f9eddb3931491aab1aeafbc1e7f1ca2a012db4.sol", "function": "Command(address,bytes)", "lineno": 29, "title": "Unchecked CALL return value", "type": "Informational"}, {"address": 1097, "code": "this.balance);              ", "debug": "", "description": "Multiple sends exist in one transaction, try to isolate each external call into its own transaction. As external calls can fail accidentally or deliberately.\nConsecutive calls: \nCall at address: 1191\n", "filename": "/data/demo_analysis/source_code/0x70f9eddb3931491aab1aeafbc1e7f1ca2a012db4.sol", "function": "GetHoneyFromJar()", "lineno": 16, "title": "Multiple Calls", "type": "Information"}, {"address": 1097, "code": "this.balance);              ", "debug": "", "description": "A possible transaction order independence vulnerability exists in function GetHoneyFromJar(). The value or direction of the call statement is determined from a tainted storage location", "filename": "/data/demo_analysis/source_code/0x70f9eddb3931491aab1aeafbc1e7f1ca2a012db4.sol", "function": "GetHoneyFromJar()", "lineno": 16, "title": "Transaction order dependence", "type": "Warning"}, {"address": 1191, "code": "fer(this.balance);\n        }     ", "debug": "SOLVER OUTPUT:\ncallvalue: 0xde0b6b3a7640001\nretval_1097: 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff\ncalldata_HomeyJar_0: 0xe0b0452100000000000000000000000000000000000000000000000000000000\ncalldatasize_HomeyJar: 0x4\n", "description": "In the function `GetHoneyFromJar()` a non-zero amount of Ether is sent to msg.sender.\nIt seems that this function can be called without restrictions.", "filename": "/data/demo_analysis/source_code/0x70f9eddb3931491aab1aeafbc1e7f1ca2a012db4.sol", "function": "GetHoneyFromJar()", "lineno": 17, "title": "Ether send", "type": "Warning"}], "success": true}